### PR TITLE
#60 - What's new

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-pulse/issues/60 |
| Decisions | Created new component to display announcements into the header. Has support for reaction (uses the new component). |
| Animated GIF | JCqrW6kwgf |

### CODE REVIEW

File: vue/components/ui/atoms/dropdown/dropdown.stories.js [hunk 1]
```diff
@@ -76,6 +76,9 @@ storiesOf("Atoms", module)
                         <b>{{ text }}</b>
                         <pre v-if="subtext">{{ subtext }}</pre>
                     </template>
+                    <template v-slot:text_3="{ item: { id, text, subtext } }">
+                        <b>Special {{ text }}</b>
+                    </template>
```
Comment (gcandal) [inline vue/components/ui/atoms/dropdown/dropdown.stories.js]: ?

File: vue/components/ui/organisms/header/header.vue [hunk 2]
```diff
@@ -38,7 +38,18 @@
...
                     v-bind:items="accountDropdownItems"
                     v-bind:visible.sync="accountDropdownVisible"
-                />
+                >
+                    <template v-slot:announcement="{ item }">
+                        <div class="announcement-dropdown-content" v-on:click="onAnnoucementsClick">
+                            <span class="announcement-dropdown-text">{{ item.text }}</span>
+                            <div
+                                class="dot"
+                                v-bind:style="{ backgroundColor: String(notifyColor) }"
```
Comment (gcandal) [inline vue/components/ui/organisms/header/header.vue]: Do this in css

File: vue/components/ui/organisms/header/header.vue [hunk 3]
```diff
@@ -38,7 +38,18 @@
                 <dropdown
                     v-bind:items="accountDropdownItems"
                     v-bind:visible.sync="accountDropdownVisible"
-                />
+                >
+                    <template v-slot:announcement="{ item }">
+                        <div class="announcement-dropdown-content" v-on:click="onAnnoucementsClick">
```
Comment (gcandal) [inline vue/components/ui/organisms/header/header.vue]: Maybe `dropdown-item-announcement`, I got a little confused about what this was and had to look at the whole HTML to understand.

File: vue/components/ui/organisms/header/header.vue [hunk 4]
```diff
@@ -337,6 +402,12 @@ export const Header = {
                 });
             }
             return items;
+        },
+        announcementsToRead() {
+            const current = Date.now();
+            return this.announcements.items.length === 0
+                ? false
+                : this.announcements.items[0].timestamp > current - this.announcements.new_delta;
```
Comment (gcandal) [inline vue/components/ui/organisms/header/header.vue]: Sort them first by timestamp, not doing so is sort of a big assumption.

File: vue/components/ui/organisms/header/header.vue [hunk 5]
```diff
@@ -301,13 +356,22 @@ export const Header = {
...
             type: Object,
             default: () => ({})
+        },
+        notifyColor: {
+            type: String,
+            default: "#3c80cd"
+        },
+        announcements: {
+            type: Object,
+            default: {}
```
Comment (gcandal) [inline vue/components/ui/organisms/header/header.vue]: Optionals -> use `null`. Semantically, it's a little bit different to have something not defined or having it defined and empty.

File: vue/components/ui/organisms/header/header.vue [hunk 6]
```diff
@@ -264,6 +290,35 @@
     font-weight: 600;
     margin: 6px 0px 0px 0px;
 }
+
+.header-ripe ::v-deep .dropdown > .dropdown-item .announcement-dropdown-text {
+    width: auto;
+}
+
+.header-ripe ::v-deep .dropdown > .dropdown-item .dot {
```
Comment (gcandal) [inline vue/components/ui/organisms/header/header.vue]: Can't you do this without the v-deep?

File: vue/components/ui/organisms/header/header.vue [hunk 7]
```diff
@@ -264,6 +290,35 @@
...
+    top: 60px;
+}
+
+body.mobile .header-ripe ::v-deep .announcements-container {
+    width: 100%;
+}
+
+body.mobile .header-ripe ::v-deep .announcements-container > .announcements {
+    height: 500px;
+}
```
Comment (gcandal) [inline vue/components/ui/organisms/header/header.vue]: Can't you do this without the v-deep?

File: vue/components/ui/organisms/header/announcement-modal.vue [hunk 8]
```diff
@@ -0,0 +1,281 @@
+<template>
+    <transition name="fade">
+        <div
+            class="announcements-container"
+            v-bind:class="{ visible: visible }"
```
Comment (gcandal) [inline vue/components/ui/organisms/header/announcement-modal.vue]: Don't think this is needed

File: vue/components/ui/organisms/header/announcement-modal.vue [hunk 9]
```diff
@@ -0,0 +1,281 @@
+<template>
+    <transition name="fade">
+        <div
+            class="announcements-container"
+            v-bind:class="{ visible: visible }"
+            v-show="isVisible"
+            v-on:click.stop
```
Comment (gcandal) [inline vue/components/ui/organisms/header/announcement-modal.vue]: Why is this needed? Can't recall.

File: vue/components/ui/organisms/header/announcement-modal.vue [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+    line-height: 18px;
+}
+
+.announcements-container .announcement-header .description {
+    font-size: 14px;
+    line-height: 14px;
+    margin: 0px 0px 10px 0px;
+}
+
+.announcements-container .announcement-header ::v-deep .checkbox .choice {
```
Comment (gcandal) [inline vue/components/ui/organisms/header/announcement-modal.vue]: No need for this, let's respect the design system.

File: vue/components/ui/organisms/header/announcement-modal.vue [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+    padding-left: 0px;
+}
+
+.announcements-container .announcement-header ::v-deep .button-icon {
```
Comment (gcandal) [inline vue/components/ui/organisms/header/announcement-modal.vue]: If you give a class to your button-icon then I think you won't need the v-deep.

File: vue/components/ui/organisms/header/announcement-modal.vue [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+}
+
+.announcements-container .announcements .announcement .content {
+    font-size: 14px;
+    line-height: 25px;
+    margin-bottom: 40px;
+    white-space: pre-line;
+}
+
+.announcements-container .announcements .announcement .link {
```
Comment (gcandal) [inline vue/components/ui/organisms/header/announcement-modal.vue]: No need, let's use the design system.

File: vue/components/ui/organisms/header/announcement-modal.vue [hunk 13]
```diff
@@ -0,0 +1,281 @@
...
+    border-color: #1d2631;
+}
+
+.announcements-container .announcements .announcement .reaction {
+    float: right;
+    line-height: initial;
```
Comment (gcandal) [inline vue/components/ui/organisms/header/announcement-modal.vue]: Please update to the newest reaction and confirm if this is needed.

File: vue/components/ui/organisms/header/announcement-modal.vue (hunk 9 above)
Comment (BeeMargarida) [reply vue/components/ui/organisms/header/announcement-modal.vue]: Prevent the announcements of closing when clicking inside the announcements modal

File: vue/components/ui/organisms/header/header.vue (hunk 4 above)
Comment (BeeMargarida) [reply vue/components/ui/organisms/header/header.vue]: fixed in header-pulse of ripe-pulse

File: vue/components/ui/organisms/header/announcement-modal.vue (hunk 13 above)
Comment (BeeMargarida) [reply vue/components/ui/organisms/header/announcement-modal.vue]: Fix in PR

**Comment by gcandal** [comment]
This got too far from `master`, closing in favor of https://github.com/ripe-tech/ripe-components-vue/pull/114 since it was easier to start from scratch based on this than merging/rebasing.
